src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -668,6 +668,18 @@ ResolveGroupShardPlacement(GroupShardPlacement *groupShardPlacement,
 }
 
 
+/*
+ * HasAnyNodes returns whether there are any nodes in pg_dist_node.
+ */
+bool
+HasAnyNodes(void)
+{
+	PrepareWorkerNodeCache();
+
+	return WorkerNodeCount > 0;
```

Comment:
I guess this logic may struggle with inactive nodes or secondary nodes, but if a user in the stage of dealing with nodes, than they should be able to fix everything easily.